django/contrib/admin/templatetags/admin_modify.py

```diff
@@ -63,7 +64,8 @@ def submit_row(context):
         'show_save_as_new': not is_popup and change and save_as,
         'show_save_and_add_another': (
             context['has_add_permission'] and not is_popup and
-            (not save_as or context['add'])
+            (not save_as or context['add']) and
+            context.get('show_save_and_add_another', True)
```

Comment:
Given this is like `show_save` and `show_save_and_continue` can you move the `get()` call here up to be consistent with those?

Reply:
moved `get()` calls for those to variables to down,
because it's more clean - these vasr are needed only once

Reply:
done in a bit different way